python/dgl/data/movielens.py

```diff
@@ -0,0 +1,644 @@
+    HAS_TORCH = False
+else:
+    HAS_TORCH = True
+
+def check_pytorch():
+    """Check if PyTorch is the backend."""
+    if not HAS_TORCH:
+        raise ModuleNotFoundError(
+            "MovieLensDataset requires PyTorch to be the backend."
+        )
+
+class MovieLensDataset(DGLDataset):
+    r"""MovieLens dataset for edge prediction tasks. The raw datasets are extracted from 
+    `MovieLens <https://grouplens.org/datasets/movielens/>`, introduced by
+    `Movielens unplugged: experiences with an occasionally connected recommender system <https://dl.acm.org/doi/10.1145/604045.604094>`.
+
+    The datasets consist of user ratings for movies and incorporate additional user/movie information in the form of features. 
+    The nodes represent users and movies, and the edges store ratings that users assign to movies.
+
+    Statistics:
+
+    MovieLens-100K (ml-100k)
+
+    - Users: 943
+    - Movies: 1,682
+    - Ratings: 100,000 (1, 2, 3, 4, 5)
+
+    MovieLens-1M (ml-1m)
+
+    - Users: 6,040
+    - Movies: 3,706
+    - Ratings: 1,000,209 (1, 2, 3, 4, 5)
+
+    MovieLens-10M (ml-10m)
+
+    - Users: 69,878 
+    - Movies: 10,677
+    - Ratings: 10,000,054 (0.5, 1, 1.5, ..., 4.5, 5.0)
+
+    Parameters
+    ----------
+    name: str
+        Dataset name. (:obj:`"ml-100k"`, :obj:`"ml-1m"`, :obj:`"ml-10m"`). 
+    valid_ratio: int
+        Ratio of validation samples out of the whole dataset. Should be in (0.0, 1.0). 
+    test_ratio: int, optional
+        Ratio of testing samples out of the whole dataset. Should be in (0.0, 1.0) for :obj:`"ml-1m"` and :obj:`"ml-10m"`. This parameter is invalid
+        when :obj:`name` is :obj:`"ml-100k"`, since its testing samples are pre-specified. 
+        Default: None
+    raw_dir : str, optional
+        Raw file directory to download/store the data.
+        Default: ~/.dgl/
+    force_reload : bool, optional
+        Whether to re-download(if the dataset has not been downloaded) and re-process the dataset. 
+        Default: False
+    verbose : bool, optional
+        Whether to print progress information. Default: True.
+    transform : callable, optional
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+    random_state : int, optional
+        Random seed used for random dataset split. Default: 0
+
+    Attributes
+    ----------
+    user_feat: torch.Tensor
+        User features
+    movie_feat: torch.Tensor
+        Movie features
+    info: dict
+        Store training, validation and testing rating pairs.
+
+    Notes
+    -----
+    - Each time when the dataset is loaded with a different setting of training, validation and test ratio, the parameter
+    :obj:`force_reload` should be set to :obj:`True`. Otherwise the previous split of the dataset will be loaded and the new split will not come into effect.
+    updated in the storage.
```

Comment:
Regarding

```
    - Each time when the dataset is loaded with a different setting of training, validation and test ratio, the parameter
    :obj:`force_reload` should be set to :obj:`True`. Otherwise the previous split of the dataset will be loaded and the new split will not come into effect.
    updated in the storage.
```

Can we handle this automatically as described [here](https://github.com/dmlc/dgl/issues/4293)? See the "Proposal" section.

Reply:
One way is to store valid_ratio and test_ratio in a separate file when saving graphs. This file would be very small and can be quickly loaded. Each time when movielens API gonna load graphs, it loads this small file at first, check whether loaded ratio match with current ratio specified by users in API arguments. If yes, directly load graphs in local disk. If not, go to process(). What do you think?

Reply:
That sounds good to me.

Reply:
fix in the latest commit. I've implemented `check_version()` to check whether the current valid_ratio and test_ratio are identical to the last setting. If not, dataset would be re-processed. The ratio setting would be stored in a local file specified by `self.version_path`. I also override `__repr__()` to correctly display information of dataset, such as current ratio and local storage path.

Reply:
Thanks. I' ll try it once we update the s3 file. Let's keep this unresolved for now.